packages/vuetify/src/styles/settings/_elevations.scss

```diff
@@ -1,101 +1,97 @@
 @use '../tools/functions' as *;
 
-$shadow-key-umbra-opacity: rgba(0, 0, 0, 0.2) !default;
-$shadow-key-penumbra-opacity: rgba(0, 0, 0, 0.14) !default;
-$shadow-key-ambient-opacity: rgba(0, 0, 0, 0.12) !default;
```

Comment:
If you do this then it doesn't need to be defined in theme unless you're customising it. 

```scss
$shadow-key-umbra-opacity: var(--v-shadow-key-umbra-opacity, rgba(0, 0, 0, 0.2)) !default;
$shadow-key-penumbra-opacity: var(--v-shadow-key-penumbra-opacity, rgba(0, 0, 0, 0.14)) !default;
$shadow-key-ambient-opacity: var(--v-shadow-key-penumbra-opacity, rgba(0, 0, 0, 0.12)) !default;
```

Reply:
Yes, Shadows are allowed to be different in each theme.

The current shadows provided by Vuetify looks accurate in both light & dark theme. However, If we change the shadow in light theme we probably need a different shadow style in dark mode (_almost 99% of the time_).

Hence, allowing different shadows based on the theme (_light, dark, coffee, high contrast_) will be easy and convenient for every developer.

Please let me know your thoughts on this.

Reply:
Yeah this wouldn't prevent that, we just wouldn't need it in the theme by default. You could still add it yourself if you want.

Reply:
Yes, you are 💯 true. However, I am theme vendor and while styling our template I reuse the shadow mixin from Vuetify.

```scss
@use "vuetify/lib/styles/tools/elevation" as elevation;

.selector {
  @include elevation.elevation(3);
}
```

Now, here's table turns. If we don't have this PR merged in vuetify then even if I add the shadow colors in my theme object I will still get the same old shadow applied to `.selector` as this will use vuetify's shadows not the one I overrode.

I hope this will clear the picture of why we need dynamic shadows. Using shadow mixin and getting most out of vuetify with DRY makes much more sense for this PR.

I hope I was able to present the need for it correctly.

Awaiting your response.

Reply:
I'm not disagreeing with you, I'm saying do that in the PR.

Reply:
Oh, sorry for the confusion. You want changes like below:
```scss
$shadow-key-umbra-opacity: var(--v-shadow-key-umbra-opacity, rgba(0, 0, 0, 0.2)) !default;
$shadow-key-penumbra-opacity: var(--v-shadow-key-penumbra-opacity, rgba(0, 0, 0, 0.14)) !default;
$shadow-key-ambient-opacity: var(--v-shadow-key-penumbra-opacity, rgba(0, 0, 0, 0.12)) !default;
```

So there can be minimal changes in the current code.

Ok, Let me update it.

Thanks for reviewing it :)